src/cpp/src/text2image/stable_diffusion_3_pipeline.hpp

```diff
@@ -0,0 +1,576 @@
+                timestep = ov::Tensor(ov::element::f32, {timestep_size});
+                float* timestep_data = timestep.data<float>();
+
+                for (size_t i = 0; i < timestep_size; ++i) {
+                    timestep_data[i] = timesteps[inference_step];
+                }
+            } else {
+                // just assign to save memory copy
+                latent_cfg = latent;
+                timestep = ov::Tensor(ov::element::f32, {1}, &timesteps[inference_step]);
+            }
+
+            ov::Tensor noise_pred_tensor = m_transformer->infer(latent_cfg, timestep);
+
+            ov::Shape noise_pred_shape = noise_pred_tensor.get_shape();
+            noise_pred_shape[0] /= batch_size_multiplier;
+            noisy_residual_tensor.set_shape(noise_pred_shape);
+
+            if (batch_size_multiplier > 1) {
+                // perform guidance
+                float* noisy_residual = noisy_residual_tensor.data<float>();
+                const float* noise_pred_uncond = noise_pred_tensor.data<const float>();
+                const float* noise_pred_text = noise_pred_uncond + noisy_residual_tensor.get_size();
+
+                for (size_t i = 0; i < noisy_residual_tensor.get_size(); ++i) {
+                    noisy_residual[i] = noise_pred_uncond[i] +
+                                        generation_config.guidance_scale * (noise_pred_text[i] - noise_pred_uncond[i]);
+                }
+            } else {
+                noisy_residual_tensor = noise_pred_tensor;
+            }
+
+            auto scheduler_step_result = m_scheduler->step(noisy_residual_tensor, latent, inference_step);
+            latent = scheduler_step_result["latent"];
+        }
+
+        float* latent_data = latent.data<float>();
+        for (size_t i = 0; i < latent.get_size(); ++i) {
+            latent_data[i] = (latent_data[i] / m_vae_decoder->get_config().scaling_factor) +
+                             m_vae_decoder->get_config().shift_factor;
+        }
+
+        return m_vae_decoder->infer(latent);
+    }
+
+private:
+    bool do_classifier_free_guidance(float guidance_scale) const {
+        return guidance_scale > 1.0;
+    }
+
+    void initialize_generation_config(const std::string& class_name) override {
+        assert(m_transformer != nullptr);
+        const auto& transformer_config = m_transformer->get_config();
+
+        size_t vae_out_channels_size = m_vae_decoder->get_config().block_out_channels.size();
+        const size_t vae_scale_factor = std::pow(2, vae_out_channels_size - 1);
+        m_transformer->set_vae_scale_factor(vae_scale_factor);
+
+        m_generation_config.height = transformer_config.sample_size * vae_scale_factor;
+        m_generation_config.width = transformer_config.sample_size * vae_scale_factor;
+
+        if (class_name == "StableDiffusion3Pipeline") {
+            m_generation_config.guidance_scale = 7.5f;
+            m_generation_config.num_inference_steps = 50;
```

Comment:
```suggestion
            m_generation_config.num_inference_steps = 28;
```
see https://github.com/huggingface/diffusers/blob/main/src/diffusers/pipelines/stable_diffusion_3/pipeline_stable_diffusion_3.py#L677